src/include/nodes/plannodes.h

```diff
@@ -76,7 +70,12 @@ typedef struct PlannedStmt
 
 	CmdType		commandType;	/* select|insert|update|delete|utility */
 
-	PlanGenerator	planGen;		/* optimizer generation */
+	/*
+	 * If the plan is done by an extension, the field below should contain
+	 * the name of the extension planner. Should be NULL if planning is done
+	 * by Postgres planner.
+	 */
+	char		*plannerName;
```

Comment:
Maybe this does not have to be dynamic? 
```suggestion
	const char		*plannerName;
```